docs/plugins/versioning/ui/components/NavbarDropdown.js

```diff
@@ -0,0 +1,88 @@
+import React, { useState, useEffect } from 'react';
+import styles from './NavbarDropdown.module.css';
+
+const CACHE_KEY = 'kargo-docs-versions';
+const CACHE_DURATION = 1000 * 60 * 30; // 30 minutes
+const githubApiUrl = 'https://api.github.com/repos/akuity/kargo/branches';
+
+function NavbarDropdown() {
+  console.log("Navbar dropdown initialized");
+  const [versions, setVersions] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [currentVersion, setCurrentVersion] = useState("");
+
+  useEffect(() => {
+    fetchVersions();
+  }, []);
+
+  const fetchVersions = async () => {
+    try {
+      console.log("Before fetching versions")
+      const response = await fetch(githubApiUrl);
+      const branches = await response.json();
+      console.log("fetched branches are: ", branches);
+
+      const releaseBranches = branches
+        .map(branch => branch.name)
+        .filter(name => /^release-1/.test(name))
+        .map(name => {
+          const [major, minor] = name.split('.');
+          return {
+            version: `${major}-${minor}`,
+            url: `https://${major}-${minor}.docs.kargo.io${window.location.pathname}`
+          };
+        });
+        console.log("These are release branches before unshifting: ", releaseBranches)
+      // Add current version
+      releaseBranches.unshift({
+        version: 'current',
+        url: `https://docs.kargo.io${window.location.pathname}`
+      });
```

Comment:
"Current" might be a poor word choice. I know you mean it as the docs for the _latest_ version, but I think will be too easily misinterpreted as the version of the docs currently selected by the dropdown.

tbh, I wouldn't bother with anything like this.

I would suggest that the title of the dropdown menu item be the _currently selected_ version and when expanded, can unconditionally show all the versions (including the currently selected one).